### PR TITLE
[APPC-3853] Integrate Offer Wall

### PR DESCRIPTION
**Pull request by AngeloMendonca**
**What does this PR do?**

   Integrates Offer Wall, or Challenge Reward, to wallet android (Payment method, Rewards view card, Transactions)

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] By commits.

**How should this be manually tested?**

  Flow on how to test this.

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-3853](https://aptoide.atlassian.net/browse/APPC-3853)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] Database migration (if applicable)?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

[APPC-3853]: https://aptoide.atlassian.net/browse/APPC-3853?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: feature/challenge-reward/data/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+dependencies {
+  implementation(project(":legacy:billing"))
+  implementation(project(":core:analytics"))
+  implementation(project(":core:network:microservices"))
+  implementation(project(":feature:wallet-info:data"))
+
+  // FYBER
+  implementation("com.fyber:fairbid-sdk:3.42.0")
```
Comment (giovanni-aptoide) [inline feature/challenge-reward/data/build.gradle.kts:26]: we have the file libs.version to do that, please move it, and you can import the rxjava bundle

File: feature/challenge-reward/data/src/main/java/com/appcoins/wallet/feature/challengereward/data/ChallengeRewardManager.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package com.appcoins.wallet.feature.challengereward.data
+
+import android.app.Activity
+import com.appcoins.wallet.feature.challengereward.data.model.ChallengeRewardListener
+import com.fyber.fairbid.ads.OfferWall
+import com.fyber.fairbid.ads.offerwall.ShowOptions
+
+const val APP_ID = "135904"
```
Comment (giovanni-aptoide) [inline feature/challenge-reward/data/src/main/java/com/appcoins/wallet/feature/challengereward/data/ChallengeRewardManager.kt:8]: whats is that? Shouldn’t it be in the buildConfig?

File: feature/challenge-reward/data/src/main/java/com/appcoins/wallet/feature/challengereward/data/ChallengeRewardManager.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    OfferWall.start(
+      activity = activity,
+      appId = APP_ID,
+      offerWallListener = ChallengeRewardListener(),
+      disableAdvertisingId = false
+    )
+  }
+
+  fun onNavigate() {
+    val customParameters = mutableMapOf("key" to "value") // TODO Personalise Parameters
```
[the remark this answers is not in the document]
Comment (AngeloMendonca) [reply feature/challenge-reward/data/src/main/java/com/appcoins/wallet/feature/challengereward/data/ChallengeRewardManager.kt:24]: True, forgot to remove that, we don't provide customParameters

File: feature/challenge-reward/data/src/main/java/com/appcoins/wallet/feature/challengereward/data/ChallengeRewardManager.kt (hunk 2 above)
Comment (AngeloMendonca) [reply feature/challenge-reward/data/src/main/java/com/appcoins/wallet/feature/challengereward/data/ChallengeRewardManager.kt:8]: Definitely Should!!! Was one of the first things that i did and totally forgot about it, sorry

File: feature/challenge-reward/data/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  namespace = "com.appcoins.wallet.feature.challengereward.data"
+
+  @Suppress("UnstableApiUsage")
+  buildFeatures {
+    compose = true
+  }
+
+  @Suppress("UnstableApiUsage")
+  composeOptions {
+    kotlinCompilerExtensionVersion = "1.4.4"
```
Comment (giovanni-aptoide) [inline feature/challenge-reward/data/build.gradle.kts:15]: it should be only defined on ExtensionPlugin

File: feature/challenge-reward/data/src/main/java/com/appcoins/wallet/feature/challengereward/data/presentation/ViewModelProvider.kt [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.appcoins.wallet.bdsbilling.repository.BdsRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class InjectionsProvider @Inject constructor(
+  val bdsRepository: BdsRepository,
+) : ViewModel()
+
```
Comment (giovanni-aptoide) [inline feature/challenge-reward/data/src/main/java/com/appcoins/wallet/feature/challengereward/data/presentation/ViewModelProvider.kt:18]: is a viewModelProvider really necessary?

**Review by rodolfobcardoso — APPROVED** [review]

**Review by rodolfobcardoso — APPROVED** [review]

**Review by rodolfobcardoso — APPROVED** [review]

**Pull request merged**